csquotes.sty

```diff
@@ -825,28 +825,41 @@
      \endgroup}
     {#1}}
 
-% \lowercase: workaround for page headers of book.cls et al.
+% The following checks if a string is completely uppercase. If so, it is lowecased,
+% if not, it is returned as is. This works around issues with uppercased language
+% names e.g. in page headers of book.cls et al., while letting mixed-case language
+% names such as USenglish untouched.
+\def\csq@lc@if@uc#1{%
+ \begingroup
+ \def\csq@tempa{#1}%
+ \def\csq@tempb{\MakeUppercase{#1}}%
+  \ifx\csq@tempa\csq@tempb
```

Comment:
isn't this test always false, given the two non equal definitions above?

Reply:
You're right. I forgot `\MakeUppercase` is not expandable.

@josephwright please revert my commit.

Reply:
ifx would be false even if it were expandable (but you could have fixed it with edef in that case)

Reply:
assuming that the names are always ascii (so the later `\lowercase` is safe)  you could do `\uppercase{\def\csq@tempa{#1}}`   then `\ifx\csq@tempa\csq@tempb`

Reply:
Yes, this seems to work.
```diff
diff --git a/csquotes.sty b/csquotes.sty
index dacb622..50a5401 100644
--- a/csquotes.sty
+++ b/csquotes.sty
@@ -832,7 +832,7 @@
 \def\csq@lc@if@uc#1{%
  \begingroup
  \def\csq@tempa{#1}%
- \def\csq@tempb{\MakeUppercase{#1}}%
+  \uppercase{\def\csq@tempb{#1}}%
   \ifx\csq@tempa\csq@tempb
      \lowercase{#1}%
   \else
```
@josephwright, interested?